server/src/main/java/io/crate/execution/dsl/projection/SourceIndexWriterReturnSummaryProjection.java

```diff
@@ -71,6 +74,7 @@ public SourceIndexWriterReturnSummaryProjection(RelationName relationName,
         sourceUri = (InputColumn) Symbols.fromStream(in);
         sourceUriFailure = (InputColumn) Symbols.fromStream(in);
         lineNumber = (InputColumn) Symbols.fromStream(in);
+        sourceParsingFailure = (InputColumn) Symbols.fromStream(in); //TODO this is draft, add version check
```

Comment:
As per comment, need to check version here.

Not sure how to proceed - just retuning null `if version < 4.7.2` would make it the only nullable `Input`here and would have to check [null](https://github.com/crate/crate/pull/12229/files#diff-1931720ac30a5f869ae7f2449bd5bd9956dbfc9581bc3b6fb8d6aae3e9ce2c0bR146) here.

Reply:
As you tagged it as a `bug`, I think `sourceParsingFailure` should be back ported. In which case you do not need version checks for streaming. 
In case where the version is lower that it is not back ported,
```
        } catch (IOException e) {
             lineProcessor.setFailure(e.getMessage());
             lineProcessor.setUriFailure(e.getMessage());
             return true;
         } catch (Exception e) {
             lineProcessor.setParsingFailure(e.getMessage());
             return true;
```

this catch blocks will also not be there and will just fail as a bug (won't reach the point, https://github.com/crate/crate/pull/12229/files#diff-53c11b5d99dc6cd6d5ea9e7a33a2f3f197c476ddd9675737246257aedbce2468R259).

Reply:
> As per comment, need to check version here. Not sure how to proceed - just retuning [null](https://github.com/crate/crate/pull/12229/files#diff-1931720ac30a5f869ae7f2449bd5bd9956dbfc9581bc3b6fb8d6aae3e9ce2c0bR146) if version < 4.7.2 would make it the only nullable Inputhere and would have to check null here.

This is resolved by your new commit https://github.com/crate/crate/pull/12229/commits/aaa55ce7979d0742fbd6b189ee652638ea8d5121 right?

Reply:
> Currently implemented by [creating new instance](https://github.com/crate/crate/pull/12229/commits/aaa55ce7979d0742fbd6b189ee652638ea8d5121)

It looks like
```
toCollect.add(SourceParsingFailureExpression.getReferenceForRelation(table.ident())); // CopyFromPlan
```
is streamed through `FileUriCollectPhase` as well. So if the ColumnIdent is recovered, `toCollect` should also be recovered(consistency wise, not that it has to in order to run successfully).

Reply:
In my opinion, having the null check seems more preferable. Could you explain why you would `new` instead?

Reply:
> In my opinion, having the null check seems more preferable. Could you explain why you would `new` instead?

Only style/consistency reasons - in order not to have mixed situation when some inputs are nullable and some are not in the `UpsertResultContext`. But on the other hand, if it legit can be null maybe it's actually better and less questionable. 

I'm currently inclined towards reverting second commit and make null check unless there is strong reason to make it non-nullable. Let's see what @mfussenegger says.